x-pack/plugins/data_visualizer/public/application/common/components/link_card/link_card.tsx

```diff
@@ -71,7 +71,8 @@ export const LinkCard: FC<LinkCardProps> = ({
             {typeof icon === 'string' ? (
               <EuiIcon size="m" type={icon} aria-label={iconAreaLabel} />
             ) : (
-              icon
+              // TODO: check this, icon might be a ComponentType
+              (icon as unknown as React.ReactNode)
```

Comment:
The type allows `ComponentType` and this would render incorrectly in runtime. Worth following up (or feel free to improve in this PR)

Reply:
Thanks for the work on this @Dosant! 🙏 

A couple of questions. Is the comment still needed?

Also, to clarify - are you saying that if the 'icon' is a component then it would not render correctly? 
I believe this component is currently only used with icon types as strings (e.g. 'pencil') passed in. Should it be updated to not allow icon components to be passed in?
cc @qn895 for anything I may have missed.

Reply:
Hi! Thanks for looking. 
> Also, to clarify - are you saying that if the 'icon' is a component then it would not render correctly?

Yes!

The type of component allows `ComponentType` but it isn't handled properly during rendering. I put a TODO to call this out. If the `ComponentType` is not intended to be used, then the easier fix would be indeed to exclude it from the props. I'll take a look

Reply:
I updated 👍